great_expectations/expectations/expectation.py

```diff
@@ -1542,24 +1536,18 @@ def _validate(
         null_count = metrics.get("column_values.nonnull.unexpected_count")
         unexpected_count = metrics.get(self.map_metric + ".unexpected_count")
 
-        success = None
         if total_count is None or null_count is None:
+            total_count = nonnull_count = 0
+        else:
+            nonnull_count = total_count - null_count
+
+        success = None
+        if total_count == 0 or nonnull_count == 0:
```

Comment:
Same comments about default `get()` to `0` and setting `success` to be a `bool` as opposed to `Optional[bool]`

Reply:
@cdkini For now, I want to keep this part intact for `_format_map_output()` to work as before (could be for another PR).